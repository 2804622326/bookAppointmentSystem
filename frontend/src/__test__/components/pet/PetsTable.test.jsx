import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import PetsTable from '../../../components/pet/PetsTable.jsx';

jest.mock('../../../components/pet/PetService.js', () => ({
  updatePet: jest.fn(() => Promise.resolve({ message: 'updated' })),
  deletePet: jest.fn(() => Promise.resolve({ message: 'deleted' })),
  addPet: jest.fn(() => Promise.resolve({ message: 'added' }))
}));

test('renders pets and handles add', async () => {
  const pets = [{ id: 1, name: 'A', type: 'Dog', breed: 'B', color: 'C', age: 2 }];
  const { asFragment } = render(
    <MemoryRouter>
      <PetsTable pets={pets} onPetsUpdate={() => {}} isEditable isPatient appointmentId={1} />
    </MemoryRouter>
  );
  fireEvent.click(screen.getAllByRole('link')[0]);
  expect(asFragment()).toMatchSnapshot();
});
