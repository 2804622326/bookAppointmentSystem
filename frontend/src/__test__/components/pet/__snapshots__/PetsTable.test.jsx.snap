// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders pets and handles add 1`] = `
<DocumentFragment>
  <section>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h4>
          Pets:
        </h4>
      </div>
      <div
        class="d-flex justify-content-end col"
      >
        <a
          class="text-info"
          href="/"
        >
          <h2>
             
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4"
              />
            </svg>
          </h2>
        </a>
      </div>
    </div>
    <table
      class="table table-sm table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Type
          </th>
          <th>
            Breed
          </th>
          <th>
            Color
          </th>
          <th>
            Age
          </th>
          <th
            colspan="3"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            A
          </td>
          <td>
            Dog
          </td>
          <td>
            B
          </td>
          <td>
            C
          </td>
          <td>
            2
          </td>
          <td>
            <button
              class="btn btn-sm btn-warning btn btn-primary"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.854.146a.5.5 0 0 0-.707 0L10.5 1.793 14.207 5.5l1.647-1.646a.5.5 0 0 0 0-.708zm.646 6.061L9.793 2.5 3.293 9H3.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.207zm-7.468 7.468A.5.5 0 0 1 6 13.5V13h-.5a.5.5 0 0 1-.5-.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.5-.5V10h-.5a.5.5 0 0 1-.175-.032l-.179.178a.5.5 0 0 0-.11.168l-2 5a.5.5 0 0 0 .65.65l5-2a.5.5 0 0 0 .168-.11z"
                />
              </svg>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</DocumentFragment>
`;
